codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameReader.java

```diff
@@ -229,28 +235,18 @@ private void processHeaderState(ByteBuf in) throws Http2Exception {
                 verifyUnknownFrame();
                 break;
         }
+        readingHeaders = false;
```

Comment:
We should be able to simplify the management of this variable: we switch back to `readingHeaders = true` as soon as we consume the frame payload, so we can set it right after we `.readSlice(..)` and before we attempt to process the payload.